lib/src/main/kotlin/com/swmansion/starknet/service/http/HttpRequest.kt

```diff
@@ -4,9 +4,8 @@ import com.swmansion.starknet.provider.Request
 import java.util.concurrent.CompletableFuture
 import java.util.function.Function
 
-// Whole HttpResponse is required for deserialization instead of just
-// body while gateway is supported, because gateway errors are returned
-// as http errors with json body.
+// Gateway errors were returned as http errors with json body.
+// Now that gateway is removed, this can be refactored to only use the response body instead of the whole HttpResponse.
 typealias HttpResponseDeserializer<T> = Function<HttpResponse, T>
```

Comment:
I think we can remove this comment altogether and stick to the current approach. 

I'm pretty sure RPC nodes can fall prey to network errors which could result in HTTP error instead of JSON-RPC one. Changing this could hinder our ability to properly handle such situation.

Current usage of `HttpResponseDeserializer`:
```kotlin
@JvmSynthetic
internal fun <T> buildJsonHttpDeserializer(deserializationStrategy: KSerializer<T>): HttpResponseDeserializer<T> {
    return Function { response ->
        if (!response.isSuccessful) {
            throw RequestFailedException(
                payload = response.body,
            )
        }

        val jsonRpcResponse =
            Json.decodeFromString(
                JsonRpcResponse.serializer(deserializationStrategy),
                response.body,
            )

        if (jsonRpcResponse.error != null) {
            throw RpcRequestFailedException(
                code = jsonRpcResponse.error.code,
                message = jsonRpcResponse.error.message,
                revertError = jsonRpcResponse.error.data?.revertError,
                payload = response.body,
            )
        }

        if (jsonRpcResponse.result == null) {
            throw RequestFailedException(message = "Response did not contain a result", payload = response.body)
        }

        jsonRpcResponse.result
    }
}
```

Reply:
>I'm pretty sure RPC nodes can fall prey to network errors which could result in HTTP error instead of JSON-RPC one. Changing this could hinder our ability to properly handle such situation.

that's true. I think we can remove the comment altogether, as well.